extensions/ql-vscode/src/test-adapter.ts

```diff
@@ -182,11 +184,33 @@ export class QLTestAdapter extends DisposableObject implements TestAdapter {
     testLogger.outputChannel.show(true);
 
     this.runningTask = this.track(new CancellationTokenSource());
+    const token = this.runningTask.token;
```

Comment:
Maybe instead of creating a `CancellationTokenSource` in the line above, you could wrap the inside of this function with `withProgress` in `commandRunner.ts`. This gives you both a progress monitor and a cancellation token.

You can use them in your calls to `removeDatabaseItem` and `openDatabase` below.